### PR TITLE
Add special type tag for numpy arrays

### LINKED ISSUE #14: Add special type for numpy arrays

### PR DESCRIPTION
**Pull request by greschd**
Fixes #14.

The reason this is not going through the entrypoint mechanism (as opposed to `sympy` types) is that `h5py` uses `numpy` anyway, so there's no _additional_ import penalty.

What might be somewhat debatable here is that `np.str_` will be cast to plain `str` when handled in a "regular" list. However, this is consistent with what is currently implemented for numeric types (matching against `numbers.Complex`).

It's possible that some numpy `dtypes` which don't have an HDF5 equivalent are still missing.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
     author='C. Frescolino',
     author_email='frescolino@lists.phys.ethz.ch',
     description=DESCRIPTION,
-    install_requires=['numpy', 'decorator', 'h5py', 'fsc.export'],
+    install_requires=['numpy', 'decorator', 'h5py', 'numpy', 'fsc.export'],
```
Comment (mskoenz) [inline setup.py]: numpy double import

**Comment by mskoenz** [comment]
no pip install .[test], using .[dev]

**Comment by mskoenz** [comment]
Coverage @ 94%

File: setup.py (hunk 1 above)
Comment (greschd) [reply setup.py]: whoops

**Comment by greschd** [comment]
> 
> 
> Coverage @ 94%

Should I activate codecov.io for this repo? See e.g. https://github.com/Z2PackDev/TBmodels/pull/67

**Comment by mskoenz** [comment]
Neat, but no, I think its fine without

**Comment by mskoenz** [comment]
nice! it seems hdf5 value serializer can manage structured numpy arrays

```
np.array([[1, 2], [4, 5]], dtype=[('age', 'i4'), ('weight', 'f4')])
```

**Comment by mskoenz** [comment]
Rest looks fine, maybe add the structured array to the test (and maybe a warning in case the inefficient version is used?)

**Comment by mskoenz** [comment]
Pull request can be merged, additional tests/warnings up to the devs discretion.

**Comment by greschd** [comment]
> 
> Rest looks fine, maybe add the structured array to the test (and maybe a warning in case the inefficient version is used?)

Hmm, not sure about the warning.. if HDF5 can not handle it natively, there really isn't a way to use the efficient one (unless it can be cast to a different type that can be handled natively).

**Review by mskoenz — DISMISSED** [review]
Looks good

**Comment by mskoenz** [comment]
Yes, that makes sense, I was more considering warnings for numpy constructions that might be too complicated to be handeled by hdf5 directly (like structured arrays, but they seem to work)

**Comment by greschd** [comment]
Yeah I'm not too sure on that.. on the one hand it would be nice to see a warning in cases where it's fixable, there are quite a few (e.g. `dtype=object`) where it's not, and having the warning would be quite a nuisance.

Neat indeed that structured types work.

**Comment by greschd** [comment]
Meta-question: Should I disable the "push to branch dismisses review" option?

It's a bit dangerous if you have external (untrusted) contributors - we could also just take an approved review as permission to merge with admin privileges.

**Comment by mskoenz** [comment]
what does it do exactly?

**Comment by mskoenz** [comment]
Ah, yes maybe would not be bad...

**Review by mskoenz — DISMISSED** [review]
New changes are fine (if travis works)

**Comment by greschd** [comment]
> 
> 
> what does it do exactly?

On push to a previously-approved PR branch, the approval is "removed". I would actually keep the option, as we still have the power to merge via admin rights. If there was an external contribution, you would want that to keep someone from changing the code after you had reviewed it.

**Comment by greschd** [comment]
So we can just (internally) treat an approval as "approval to merge via admin rights".

**Comment by mskoenz** [comment]
makes sense, also step your travis game up :P

**Comment by greschd** [comment]
yeah.. ended up trying to add the coverage anyway

**Comment by greschd** [comment]
> 
> 
> makes sense, also step your travis game up :P

'twas my bash game that is lacking

**Review by mskoenz — COMMENTED** [review]
Bash game now up

**Review by mskoenz — APPROVED** [review]

**Comment by mskoenz** [comment]
delete branch when done

**Pull request merged**